crates/hotshot/types/src/stake_table.rs

```diff
@@ -40,4 +47,123 @@ impl<K: SignatureKey> StakeTableEntry<K> {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+#[inline]
+/// A helper function to compute the quorum threshold given a total amount of stake.
+pub fn one_honest_threshold(total_stake: U256) -> U256 {
+    total_stake / U256::from(3) + U256::from(1)
+}
+
+#[inline]
+fn u256_to_field(amount: U256) -> CircuitField {
+    let amount_bytes: [u8; 32] = amount.to_le_bytes();
+    CircuitField::from_le_bytes_mod_order(&amount_bytes)
+}
+
+impl<TYPES: NodeType> std::iter::IntoIterator for HSStakeTable<TYPES> {
+    type Item = PeerConfig<TYPES>;
+    type IntoIter = std::vec::IntoIter<PeerConfig<TYPES>>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.0.into_iter()
+    }
+}
+
+impl<TYPES: NodeType> HSStakeTable<TYPES> {
+    pub fn commitment(&self, stake_table_capacity: usize) -> anyhow::Result<StakeTableState> {
+        if stake_table_capacity < self.0.len() {
+            return Err(anyhow::anyhow!(
+                "Stake table over capacity: {} < {}",
+                stake_table_capacity,
+                self.0.len(),
+            ));
+        }
+        let padding_len = stake_table_capacity - self.0.len();
+        let mut bls_preimage = vec![];
+        let mut schnorr_preimage = vec![];
+        let mut amount_preimage = vec![];
+        let mut total_stake = U256::from(0);
+        for peer in &self.0 {
+            bls_preimage.extend(peer.stake_table_entry.public_key().to_fields());
+            schnorr_preimage.extend(peer.state_ver_key.to_fields());
+            amount_preimage.push(u256_to_field(peer.stake_table_entry.stake()));
+            total_stake += peer.stake_table_entry.stake();
+        }
+        bls_preimage.resize(
+            <TYPES::SignatureKey as ToFieldsLightClientCompat>::SIZE * stake_table_capacity,
+            CircuitField::default(),
+        );
+        // Nasty tech debt
+        schnorr_preimage.extend(
+            std::iter::repeat_n(TYPES::StateSignatureKey::default().to_fields(), padding_len)
+                .flatten(),
+        );
+        amount_preimage.resize(stake_table_capacity, CircuitField::default());
+        let threshold = u256_to_field(one_honest_threshold(total_stake));
+        Ok(StakeTableState {
+            bls_key_comm: VariableLengthRescueCRHF::<CircuitField, 1>::evaluate(bls_preimage)
+                .unwrap()[0],
+            schnorr_key_comm: VariableLengthRescueCRHF::<CircuitField, 1>::evaluate(
+                schnorr_preimage,
+            )
+            .unwrap()[0],
+            amount_comm: VariableLengthRescueCRHF::<CircuitField, 1>::evaluate(amount_preimage)
+                .unwrap()[0],
+            threshold,
+        })
+    }
+
+    pub fn total_stakes(&self) -> U256 {
+        self.0
+            .iter()
+            .map(|peer| peer.stake_table_entry.stake())
+            .sum()
+    }
+
+    pub fn iter(&self) -> std::slice::Iter<'_, PeerConfig<TYPES>> {
+        self.0.iter()
+    }
```

Comment:
similarly, thanks to `Deref`, this should be unnecessary?

Reply:
done in [ae5454d](https://github.com/EspressoSystems/espresso-network/pull/3233/commits/ae5454dcb5ba9c67dc7aeda32459284df531c5d4)